File: C/Assignment9/Program3.c
/////////////////////////////////////////////////////////////////////////////////////////
//
// File name :   Program3.c
// Description : Program to which accept number from user & count frequency of 2 in it.
// Author :      Pallavi Omprakash Malewar
// Date :        20/06/2025
//
////////////////////////////////////////////////////////////////////////////////////////

/*
Input : 2395
Output : 1

Input : 1018
Output : 0

Input : 9000
Output : 0

Input : 922432
Output : 3
*/

#include<stdio.h>

int CountTwo(int iNo)
{
    int iDigit = 0 , iCnt = 0 ;

    while(iNo != 0)
    {
        iDigit = iNo % 10;
        if(iDigit == 2)
        {
            iCnt++;
        }
        iNo = iNo / 10;
    }
    return iCnt;
}

int main()
{
    int iValue = 0;
    int iRet = 0;

    printf("Enter Number : ");
    scanf("%d",&iValue);

    iRet = CountTwo(iValue);

    printf("2 came %d times.",iRet);

    return 0;
}

File: JAVA/Assignment 1/Program3.java
/////////////////////////////////////////////////////////////////////////////////////////
//                                                                                     //
// Problem Statement : Write a program which accept number from user and count         //
//                     frequency of 2.                                                 //
// Author : Pallavi Omprakash Malewar                                                  //
// Date : 30 Jun 2026                                                                  //
//                                                                                     //
/////////////////////////////////////////////////////////////////////////////////////////

/*

    Input: 2395
    Output : 1

    Input: 1018
    Output : 0

    Input: 9000
    Output : 0

    Input: 922432
    Output : 3

*/


import java.util.Scanner;

class Program3
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.print("Enter the Number : ");
        int No = sobj.nextInt();
        
        int freq = CountTwo(No);

        System.out.println(freq);
        
    }

    public static int CountTwo(int No)
    {
        int Digit = 0, Count = 0;

        if(No < 0)
        {
            No = -No;
        }

        if(No == 0)
        {
            return 0;
        }

        while(No != 0)
        {
            Digit = No % 10;
            if(Digit == 2)
            {
                Count++;
            }
            No /= 10;
        }
        return Count;
    }
}